osu.Framework/Audio/Track/Track.cs

```diff
@@ -124,7 +124,7 @@ public virtual double Rate
 
         public bool IsReversed => Rate < 0;
 
-        public override bool HasCompleted => IsLoaded && !IsRunning && CurrentTime >= Length;
+        public override bool HasCompleted => IsLoaded && !IsRunning && (!IsReversed ? CurrentTime >= Length : 0 <= Length);
```

Comment:
```suggestion
        public override bool HasCompleted => IsLoaded && !IsRunning && (IsReversed ? CurrentTime <= 0 : CurrentTime >= Length);
```
The negation feels weird and the check if a reversed track completed seems plain wrong?